Split up cub-RadixSortPairs.cu to parallelize compilation

TovlyFB opened the pull request:
Summary: `cub-RadixSortPairs.cu` has slow compilation times, especially on Windows. These changes split up the file into smaller components to allow each component to compile in parallel. On Windows, I observed a compile time drop from about 20 minutes to 6 minutes.

Differential Revision: D70539649

Skylion007 commented on aten/src/ATen/cuda/cub-RadixSortPairs.cuh:
```diff
@@ -1,3 +1,5 @@
+#pragma once
```
Just adding pragma once doesn't fix the compilation speed, right?

Skylion007 commented:
If we want to generalize this, since we are on newer CUDA versions now we may want to consider enabling link time optimization to NVCC which parallelize arch code generation automatically for us when we specify it.

suo commented:
@pytorchbot merge

suo commented:
@TovlyFB looks like lint is still broken, can you look

TovlyFB commented:
> @TovlyFB looks like lint is still broken, can you look

@suo The lint looks like it's complaining about some generated or included code `usr/include/c++/11/cmath` not touched in this PR. Any idea how I could debug, or maybe a rebase would help?

TovlyFB commented:
@suo rebase doesn't seemed to have fixed it

suo commented:
@malfet, these clang-tidy errors are pretty weird, do you know what might be happening here?

eqy commented:
Since it's just the macros being moved around and those are host-functions maybe a hail mary would be to add `__host__` to the macro?

TovlyFB commented:
@eqy 

> Since it's just the macros being moved around and those are host-functions maybe a hail mary would be to add `__host__` to the macro?

Just tried this but getting the same error (and some new ones)

malfet commented:
@pytorchbot merge -f "Let's test it in trunk"

ZainRizvi commented:
@pytorchbot revert -c ignoredsignal -m "Breaks lint in trunk [GH job link](https://github.com/pytorch/pytorch/actions/runs/13845459825/job/38742803351) [HUD commit link](https://hud.pytorch.org/pytorch/pytorch/commit/9a3d26cfcdb1c1be84a04baa3ee554dbe67cb049)"

TovlyFB commented:
Attempting to fix linter error by narrowing clangtidy (will move to separate PR if it works)

TovlyFB committed:
Split up cub-RadixSortPairs.cu to parallelize compilation (#148936)

Summary:

`cub-RadixSortPairs.cu` has slow compilation times, especially on Windows. These changes split up the file into smaller components to allow each component to compile in parallel. On Windows, I observed a compile time drop from about 20 minutes to 6 minutes.

Reviewed By: suo

Differential Revision: D70539649

malfet commented:
@pytorchbot merge -i
